joomla/joomla-cms
# Clean up ContentViewCategory

Pull Request for Issue # .

### Summary of Changes
This PR makes some clean up to ContentViewCategory classes:

1. Use namespace classes instead of it's alias
2. Move $dispatcher = JEventDispatcher::getInstance(); out of the loop, so we get a bit performance improvement
3. Remove code to set view layout as it is set on parent class code already https://github.com/joomla/joomla-cms/blob/staging/libraries/src/MVC/View/CategoryView.php#L212-L223

4. Remove some un-used variables (like $id, $pathway), reuse-existing variable (no need for $menu = $menu->getActive() when we have $active variable keep active menu item before)


### Testing Instructions
Code review should be enough. For human testing, please create a menu item to display articles from a category (Category Blog layout menu option for example), check and make sure articles still being displayed like before
